lib/pbench/agent/results.py

```diff
@@ -18,49 +18,55 @@ def __init__(self, result_dir, target_dir, user, prefix, config, logger):
         assert (
```

Comment:
I'd like to see docstrings here, describing the parameters. And while I know you're mostly changing and reorganizing code, it'd also be great to add type hints.

Reply:
Sure :)